src/main/java/org/apache/maven/buildcache/MojoParametersListener.java

```diff
@@ -77,8 +77,4 @@ public void afterExecutionFailure(MojoExecutionEvent event) {
     public Map<String, MojoExecutionEvent> getProjectExecutions(MavenProject project) {
         return projectExecutions.get(project);
     }
-
-    public void remove(MavenProject project) {
```

Comment:
If I understood well, this was used to "reset" the project mojo's execution spying after a "rollback" from using the cache to a regular build execution.
I'm not anymore in a position where I need this function so I deleted it. But I could use your careful reviewing on it @maximilian-novikov-db (I see you name on the oldest commit of the class :p )